### PR TITLE
refactor: Move OS definitions into yaml files

### PR DESCRIPTION
**Pull request by laf**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you signed the [Contributors agreement](http://docs.librenms.org/General/Contributing/)
- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

Refactored OS definitions into yaml files so we aren't loading everything in `$config`.

Some things for now:

1. Polling and discovery should function as normal but over time we should move some of the grunt of these into functions which use data from the yaml files overall making it easier to add support for new OS.

2. We have to load all yaml files for the webui at present as we access various `$config['os'][$os]` values all over the show.

### CODE REVIEW

**Comment by murrant** [comment]
What is the reason you don't remove everything from definitions.inc.php?
Also, why the change to .gitignore?

**Comment by laf** [comment]
I'll remove everything from definitions when it's ready for merging, I'm rebasing when we add support and I do it with a quick script to pull out what's in definitions.

.gitignore is because I use composer install for yaml repo which then installs dependencies into lib/yaml/vendor

**Comment by murrant** [comment]
oh yeah, I changed the vendor in my .gitignore to /vendor for my snmp-object branch.

File: tests/YamlTest.php [hunk 1]
```diff
@@ -37,7 +37,13 @@ public function testYaml()
 
         $pattern = $config['install_dir'] . '/includes/definitions/*.yaml';
         foreach (glob($pattern) as $file) {
-            $data = Yaml::parse(file_get_contents($file));
+            try {
+                $data = Yaml::parse(file_get_contents($file));
+            } catch (ParseException $e) {
+                //$this->assertNotEmpty($data, "$file Could not be parsed");
+                $this->assertEmpty($e->getMessage(), "$file Could not be parsed");
```
Comment (murrant) [inline tests/YamlTest.php]: Perhaps fail another way here.
`throw new PHPUnitException("$file Could not be parsed");`
maybe...

**Comment by murrant** [comment]
Can either merge this before or after #5216 
If we merge it after, we won't have to move the library files around.  (and you can test the new dependency docs) :)

**Comment by laf** [comment]
I'm happy either way but this will need another rebase anyway after the next mib update.

Your PR is quite big though so checking might take a while :)

**Comment by laf** [comment]
Changed the web ui to no longer load all yaml files and only those the user has in the devices table. I.e if all devices are ios then only that file is loaded.

**Pull request merged**